### PR TITLE
Create MatomoTagManager component

### PR DESCRIPTION
**Pull request by bddjong**
- [x] Add matomo snippet
- [x] Ensure both matomo and siteimprove are loaded before Open Forms is ever loaded (this is required for Open Forms' internal supported analytics to detect matomo and siteimprove)

### CODE REVIEW

**Comment by AliKdhim87** [comment]
- [x] Add the Matomo Tag Manager script
- [x] Use the `beforeInteractive` script option to ensure that the script is loaded before any Next.js code execution and page hydration takes place. https://nextjs.org/docs/app/building-your-application/optimizing/scripts#strategy

> Note: Based on the provided code snippets from Utrecht, we can only enable the Matomo Tag Manager.
If we need to enable the Matomo tracker script, we can use the [matomo-next](https://github.com/SocialGouv/matomo-next?tab=readme-ov-file) package. However, we still need some information such as the site ID.

File: apps/pdc-frontend/src/components/MatomoTagManager.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <Script
+      id="_next-matomo"
+      src="https://stats.utrecht.nl/analytics/js/container_Ku3GHbr3.js"
+      nonce={nonce}
+      dangerouslySetInnerHTML={{
+        __html: `
+        // initializes the Matomo Tag Manager array
+        const _mtm = window._mtm = window._mtm || [];
+        _mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
+        `,
```
Comment (bddjong) [inline apps/pdc-frontend/src/components/MatomoTagManager.tsx:17]: Hier moeten we denk ik iets anders op verzinnen. Volgens de HTML spec 

> If there is a [src](https://www.w3.org/TR/2011/WD-html5-author-20110809/the-script-element.html#attr-script-src) attribute, the element must be either empty or contain only [script documentation](https://www.w3.org/TR/2011/WD-html5-author-20110809/the-script-element.html#inline-documentation-for-external-scripts) that also matches [script content restrictions](https://www.w3.org/TR/2011/WD-html5-author-20110809/the-script-element.html#restrictions-for-contents-of-script-elements).

https://www.w3.org/TR/2011/WD-html5-author-20110809/the-script-element.html

Zouden we het in `onLoad=` ([docs](https://nextjs.org/docs/pages/building-your-application/optimizing/scripts#executing-additional-code)) kunnen zetten?

File: apps/pdc-frontend/src/components/MatomoTagManager.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+'use clint';
```
Comment (bddjong) [inline apps/pdc-frontend/src/components/MatomoTagManager.tsx:1]:
```suggestion
'use client';
```

File: apps/pdc-frontend/src/components/MatomoTagManager.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+interface MatomoTagManagerProps extends ScriptProps {}
+
+export const MatomoTagManager = ({ nonce, ...restProps }: MatomoTagManagerProps) => {
+  return (
+    <Script
+      id="_next-matomo"
+      src="https://stats.utrecht.nl/analytics/js/container_Ku3GHbr3.js"
```
Comment (bddjong) [inline apps/pdc-frontend/src/components/MatomoTagManager.tsx:22]: Deze moeten we eigenlijk in een env var zetten bedenk ik me. De container verschilt namelijk per omgeving volgens Joram

Comment (AliKdhim87) [reply apps/pdc-frontend/src/components/MatomoTagManager.tsx:22]: Bedoel je dat we de volledige link, inclusief `container_Ku3GHbr3.js`, als een env variable moeten maken?
Omdat ik denk dat deze `container_Ku3GHbr3.js` iets anders zou kunnen zijn.

Comment (bddjong) [reply apps/pdc-frontend/src/components/MatomoTagManager.tsx:22]: Eigenlijk alleen de container code inderdaad.

**Review by bddjong — COMMENTED** [review]
👍 looks good @AliKdhim87 

Cant approve because I opened the PR but I'd say ship it and lets test this

**Pull request merged**